bubblewrap.c

```diff
@@ -2986,6 +2996,16 @@ main (int    argc,
       /* Ignore res, if e.g. the child died and closed child_wait_fd we don't want to error out here */
       close (child_wait_fd);
 
+      close (command_pid_sockets[1]);
+      command_pid = read_pid_from_socket (command_pid_sockets[0]);
+      close (command_pid_sockets[0]);
+
+      if (opt_json_status_fd != -1)
+        {
+          cleanup_free char *output = xasprintf ("{ \"command-pid\": %i }\n", command_pid);
+          dump_info (opt_json_status_fd, output, TRUE);
+        }
```

Comment:
This will make the uppermost process (the monitor process, outside the sandbox) block until the final process sends its pid. And, if the child process (reported as `child-pid`) fails for any reason, then `read_pid_from_socket()` will fail with a fatal error, preventing `monitor_child()` from relaying the true exit status of the child.

I think it would be better to monitor `command_pid_sockets[0]` in `monitor_child()`, and use non-blocking I/O for that, so that we output the information when we have it, but we don't wait for it.